modules/similarity-algorithms/pages/cosine-similarity.adoc

```diff
@@ -0,0 +1,95 @@
+= Cosine Similarity
+:stem: latex
+:experimental:
+
+Algorithm link: link:https://github.com/tigergraph/gsql-graph-algorithms/tree/277349ce0414ba797edcad1488b6caf1904d5beb/algorithms/Similarity/approximate_nearest_neighbors[Approximate Nearest Neighbors]
+
+
+Cosine similarity measures the cosine of the angle between two non-zero vectors of an inner product space.
+
+[stem]
+++++
+\[
+\text{Cosine Similarity} = \frac{\sum_{i=1}^{n}A_i \times B_i}{\sqrt{\sum_{i=1}^{n}A_i^2} \times \sqrt{\sum_{i=1}^{n}B_i^2}}
+\]
+++++
+
+This is commonly used to determine the similarity between two vectors, often used in text analysis and other vector space models.
+
+The algorithm can take vectors denoted by `ListAccum` or standard `vector`.
+
+== Notes
+
+This algorithm is implemented as a user-defined function.
+You need to follow the steps in xref:3.10.1@gsql-ref:querying:func/query-user-defined-functions.adoc[Add a User-Defined Function] to add the function to GSQL.
+After adding the function, you can call it in any GSQL query in the same way as a built-in GSQL function.
+
+== Specifications
+
+....
+tg_cosine_similarity_accum(A, B)
+tg_cosine_similarity(A, B)
```

Comment:
```suggestion
tg_similarity_accum(A, B, "COSINE")
```

Reply:
Changed this for both Cosine and Jaccard.